schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/model/cart/PaymentOption.java

```diff
@@ -0,0 +1,158 @@
+import com.jasify.schedule.appengine.model.payment.PaymentTypeEnum;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author wszarmach
+ * @since 24/09/15.
+ */
+public class PaymentOption implements Serializable {
+    private PaymentTypeEnum paymentType;
+    private Double fee;
+    private String feeReason;
+    private boolean enabled;
+    private String disabledReason;
+
+    private PaymentOption() {
+    }
+
+    private static String formatNumber(BigDecimal number, int precision) {
+        number = number.setScale(2, BigDecimal.ROUND_DOWN);
+        DecimalFormat df = new DecimalFormat();
+        df.setMaximumFractionDigits(precision);
+        df.setMinimumFractionDigits(precision);
+        df.setGroupingUsed(false);
+        return df.format(number);
+    }
+
+    public PaymentTypeEnum getPaymentType() {
+        return paymentType;
+    }
+
+    public void setPaymentType(PaymentTypeEnum paymentType) {
+        this.paymentType = paymentType;
+    }
+
+    public String getFeeReason() {
+        return feeReason;
+    }
+
+    public void setFeeReason(String feeReason) {
+        this.feeReason = feeReason;
+    }
+
+    public Double getFee() {
+        return fee;
+    }
+
+    public void setFee(Double fee) {
+        this.fee = fee;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public String getDisabledReason() {
+        return disabledReason;
+    }
+
+    public void setDisabledReason(String disabledReason) {
+        this.disabledReason = disabledReason;
+    }
+
+    public static class CashPaymentOption extends PaymentOption {
+        private static final String DisabledReason = "Not supported by supplier";
+
+        public CashPaymentOption() {
+            setPaymentType(PaymentTypeEnum.Cash);
+            setDisabledReason(DisabledReason);
+            setEnabled(false);
+            setFee(BigDecimal.ZERO.doubleValue());
+        }
+    }
+
+    public static class InvoicePaymentOption extends PaymentOption {
+        private static final Logger Log = LoggerFactory.getLogger(InvoicePaymentOption.class);
+        private static final long AllowedDays = 7;
+        private final static long InvoiceDaysBeforePaymentRequiredInMillis = 86400000 * AllowedDays;
+        private static final String DisabledReason = "For Invoice payments a minimum of " + AllowedDays + " days is required before the first activity";
+        private static final String FeeReason = createFeeReason();
+
+        public InvoicePaymentOption(List<ShoppingCart.Item> items) {
+            setPaymentType(PaymentTypeEnum.Invoice);
+            setDisabledReason(DisabledReason);
+            setEnabled(calculateEnabled(items));
```

Comment:
I thought we said we would disable this for now...

Quoting yourself: Removed restriction on Invoice payments must be 7 days before first activity